Example/GiniVision/ComponentAPIAnalysisViewController.swift

```diff
@@ -42,60 +50,27 @@ class ComponentAPIAnalysisViewController: UIViewController {
         // 2. Create the analysis view controller
         guard let document = document else { return }
         
-        // In case that they view is loaded but is not analysing (i.e: user import a PDF with Open With feature), it should start.
-        if !AnalysisManager.sharedManager.isAnalyzing {
-            AnalysisManager.sharedManager.analyzeDocument(withData: document.data, cancelationToken: CancelationToken(), completion: nil)
-        }
-        
         contentController = AnalysisViewController(document)
         
         // 3. Display the analysis view controller
         displayContent(contentController)
     }
     
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        
-        guard let navController = navigationController else { return }
-        if isFirstViewController(inNavController: navController) {
-            navigationItem.leftBarButtonItem = UIBarButtonItem(title: "Abbrechen", style: .plain, target: self, action: #selector(closeAction))
-        }
-    }
-    
-    func closeAction() {
-        dismiss(animated: true, completion: nil)
-    }
-    
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
+        delegate?.didAppear()
         
-        // Subscribe to analysis events which will be fired when the analysis process ends.
-        // Either with a valid result or an error.
-        NotificationCenter.default.addObserver(self, selector: #selector(handleAnalysis(errorNotification:)), name: NSNotification.Name(rawValue: GINIAnalysisManagerDidReceiveErrorNotification), object: nil)
-        NotificationCenter.default.addObserver(self, selector: #selector(handleAnalysis(resultNotification:)), name: NSNotification.Name(rawValue: GINIAnalysisManagerDidReceiveResultNotification), object: nil)
-        
-        // Because results may come in during view controller transition,
-        // check for already existent results in shared analysis manager.
-        handleExistinResults()
-        
-        // Start loading animation.
         (contentController as? AnalysisViewController)?.showAnimation()
     }
     
-    override func viewWillDisappear(_ animated: Bool) {
-        super.viewWillDisappear(animated)
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
         
-        // Never forget to remove observers when you support iOS versions prior to 9.0.
-        NotificationCenter.default.removeObserver(self)
-    }
-    
-    override func didMove(toParentViewController parent: UIViewController?) {
-        super.didMove(toParentViewController: parent)
-        
-        // Cancel analysis process to avoid unnecessary network calls.
-        if parent == nil {
-            AnalysisManager.sharedManager.cancelAnalysis()
+        if isMovingFromParentViewController || isBeingDismissed {
```

Comment:
It's nice that you removed the didMove method! Are you sure the viewDidDisappear is enough? It will be called even when a new view controller is pushed on top.

Reply:
Yes, but since it is the last view controller before the results, it won't be a problem.